eng/native/version/_version.c

```diff
@@ -0,0 +1,4 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+static char sccsid[] __attribute__((used)) = "@(#)No version information produced";
```

Comment:
when msbuild is present, our normal SCCID looks like:
```sh
$ strings artifacts/bin/coreclr/OSX.x64.Debug/libcoreclr.dylib | grep '@(#)'
@(#)Version 7.0.21.45601 @Commit: b7e10374429b5af32debc4660aa9bce41cfb903e
```
in case if the `.git/` (directory) is not present (e.g. in case of building sources from extracted achieve), it looks like:
```sh
@(#)Version 7.0.21.45601 @Commit: N/A
```
I think we can emit this line from script more efficiently in case when MSBuild is not available:
```sh
COMMIT="${$(git rev-parse HEAD 2>/dev/null):-N/A}"
printf 'static char sccsid[] __attribute__((used)) = "@(#)Version N/A @Commit: %s";\n' "$COMMIT" > artifacts/obj/_version.c
```

Reply:
The above behavior, which changes based on availability of tools, is true when `OfficialBuildId` is set. For local build we have yet another representation; with a  placeholder version in SCCSID and without commit. I have aligned it in https://github.com/dotnet/arcade/pull/7930.